js/index8.js

```diff
@@ -548,15 +548,647 @@ console.log(uniqueSortedCourses); // ['biologie', 'informatică', 'limba englez
+console.table(getCarsWithType(cars, "sedan"));
+
+/*
+Example 6 - The find method
+*/
+
+console.log(`-Example 6 - The find method
+-`);
+
+const getCarByModel = (cars, model) => {
+  return cars.find((x) => x.model === model);
+};
+
+console.log(getCarByModel(cars, "F-150"));
+console.log(getCarByModel(cars, "CX-9"));
+
+/*
+Example 7 - Sort method
+Let the `sortByAscendingAmount` function return a new array of cars
+sorted in ascending order by the value of the `amount` property.
+*/
+
+console.log(`-Example 7 - Sort method
+-`);
+
+const sortByAscendingAmount = (cars) => {
+  return [...cars].sort((x, y) => x.amount - y.amount);
+};
+
+console.table(sortByAscendingAmount(cars));
+
+/*
+Example 8 - Sort method
+Let the `sortByDescendingPrice` function return a new array of cars
+sorted in descending order by the value of the `price` property.
+*/
+
+console.log(`-Example 8 - Sort method
+-`);
+
+const sortByDescendingPrice = (cars) => {
+  return [...cars].sort((x, y) => y.price - x.price);
+};
+
+console.table(sortByDescendingPrice(cars));
+
+/*
+Example 9 - Sort method
+Let the `sortByModel` function return a new array of cars sorted
+by model name in alphabetical and reverse alphabetical order, depending on
+the value of the `order` parameter.
+*/
+
+console.log(`-Example 9 - Sort method
+-`);
+
+const sortByModel = (cars, order) => {
+  // return a.sortValue?.[1].localeCompare(b.sortValue?.[1]);
+  let newCars;
+  if (order === "asc") {
+    newCars = [...cars].sort((x, y) => x.make.localeCompare(y.make));
+  } else if (order === "desc") {
+    newCars = [...cars].sort((x, y) => y.make.localeCompare(x.make));
+  }
+
+  return newCars;
+};
+
+console.table(sortByModel(cars, "asc"));
+console.table(sortByModel(cars, "desc"));
+
+/*
+Example 10 - Reduce method
+Let the `getTotalAmount` function return the total number of cars (the value
+of the `amount` properties).
+*/
+
+console.log(`-Example 10 - Reduce method
+-`);
+
+// const countLikes = (tweetss) => {
+//   return tweetss.reduce((totalLikes, tweet) => totalLikes + tweet.likes, 0);
+// };
+
+const getTotalAmount = (cars) => {
+  return cars.reduce((acc, value) => acc + value.amount, 0);
+};
+
+console.log(getTotalAmount(cars));
+
+/*
+Example 11 - Method chains
+Let the `getAvailableCarNames` function return an array of car models, but 
+only those that are currently for sale.
+*/
+
+console.log(`-Example 11 - Method chains
+-`);
+
+const getModelsOnSale = (cars) => {
+  let carModel;
+  carModel = cars.filter((x) => x.onSale === true).map((x) => x.model);
+  return carModel;
+};
+console.table(getModelsOnSale(cars));
+
+/*
+Example 12 - Method chains
+Let the `getSortedCarsOnSale` function return an array of cars for sale 
+(onSale property), sorted by price ascending.
+*/
+
+console.log(`-Example 12 - Method chains
+-`);
+
+const getSortedCarsOnSale = (cars) => {
+  return cars
+    .filter((x) => x.onSale === true)
+    .sort((a, b) => a.price - b.price);
+};
```

Comment:
no need for === true, other than that it looks good

Reply:
copied code from above :))